### ISSUE TITLE
Date::Holidays::DE holidays method returns unconforming data structure

### ISSUE THREAD

**Issue by jonasbn** [jonasbn/perl-date-holidays#29]
# Issue Template

### Description

Date::Holidays::DE's holiday method returns a reference to an array instead of the _de facto_ reference to a hash.

# Bugs and Issues

### Specify what version was used

Date::Holidays::DE 1.9

### Expected behaviour

A reference to a hash is returned:

### Actual and observed behaviour

A reference to an array is returned

### Steps to reproduce the behaviour

```perl
use Data::Dumper;
use Date::Holidays::DE;
my $holidays = Date::Holidays::DE::holidays();

print STDERR Dumper $holidays;
```

```perl
$VAR1 = [
          '1514761200',
          '1522360800',
          '1522620000',
          '1525125600',
          '1525903200',
          '1526853600',
          '1538517600',
          '1545692400',
          '1545778800'
        ];
```

**Comment by jonasbn** [comment]
This fix can use the code implemented in regard to issue #18, both `format` as and helper method

**Comment by jonasbn** [comment]
Changed the structure returned from Date::Holidays::DE from a reference to an array to a reference to a hash, which is easier to work with and seems to be the defacto standard

This is all done in the adapter
